doc/api/assert.md

```diff
@@ -7,6 +7,9 @@
 The `assert` module provides a simple set of assertion tests that can be used to
 test invariants.
 
+For more information about the used equality comparisons see
+[MDN's guide on equality comparisons and sameness][mdn-equality-guide].
```

Comment:
Is this unrelated? Or is this doc change due to the change in the code?

Reply:
I removed the `Caveats` part as there are non left but I felt like this would still be a valuable information from that part, so I moved it to the top.